src/Algolia.Search.Test/EndToEnd/Recommendation/RecommendationClientTest.cs

```diff
@@ -50,7 +51,18 @@ public void TestRecommendationClient()
                 },
                 0
             );
-            Assert.DoesNotThrow(() => BaseTest.RecommendationClient.SetPersonalizationStrategy(request));
+            try
+            {
+                BaseTest.RecommendationClient.SetPersonalizationStrategy(request);
+            }
+            catch (AlgoliaApiException e)
```

Comment:
nit: Don't hesitate to use the latest C# feature such as `catch when`. Here is an example from the code-base

Reply:
Thanks for that :)